.github/workflows/windows-build.yml

```diff
@@ -0,0 +1,36 @@
+name: Windows build
+
+on:
+  pull_request:
+
+# Cancel previous runs if a more recent commit is pushed.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
+
+permissions: read-all
+
+jobs:
+  windows-build:
+    name: Build and run unit tests on Windows
+    runs-on: windows-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: '0'
+      - name: Switch to pull request branch and clone submodules
+        run: |
+          git checkout ${GITHUB_SHA}
+          git submodule update --init --recursive
+      - name: Build
+        run: |
+          mkdir build
+          cd build
+          cmake -G "Visual Studio 16 2019" -A x64 -DSPIRV_REFLECT_BUILD_TESTS=ON
```

Comment:
From https://github.com/actions/runner-images/blob/main/images/win/Windows2022-Readme.md it looks like this should be Visual Studio 2022, not 2019